test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -446,7 +468,10 @@ func runtimeHookTestHandler(ctx context.Context, c client.Client, namespace, clu
 		if err := c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: clusterName}, cluster); err != nil {
 			return err
 		}
-		if !(conditions.GetReason(cluster, clusterv1.TopologyReconciledCondition) == clusterv1.TopologyReconcileFailedReason) {
+
+		// Check for the existence of the condition if withTopologyReconciledCondition is true.
+		if withTopologyReconciledCondition &&
```

Comment:
Q: is this for checking we are blocking? can we check for a specific reason vs check for everything except failures?

Reply:
This is just to check the condition has been supplied i.e. the cached cluster is updated with the information from the hook call. We check for actual blocking behaviour in the Consistently() call in L780